packages/react-native-worklets/apple/worklets/apple/AnimationFrameQueue.mm

```diff
@@ -34,7 +38,14 @@ - (instancetype)init
 {
   AssertJavaScriptQueue();
   if constexpr (worklets::StaticFeatureFlags::getFlag("IOS_DYNAMIC_FRAMERATE_ENABLED")) {
-    bool supportsProMotion = [UIScreen mainScreen].maximumFramesPerSecond > 60;
+    bool supportsProMotion = false;
+#if TARGET_OS_OSX
+    NSScreen *screen = [NSScreen mainScreen];
+    supportsProMotion = (screen.maximumFramesPerSecond > 60);
+#else
+    supportsProMotion = [UIScreen mainScreen].maximumFramesPerSecond > 60;
+#endif // TARGET_OS_OSX
```

Comment:
Since we can't control dynamic changes of the framerate on macOS:

<img width="573" height="92" alt="Image" src="https://github.com/user-attachments/assets/c89f34a6-9dcf-45a3-bb05-79186320c760" />

lets just disable it

```suggestion
#if !TARGET_OS_OSX
    supportsProMotion = [UIScreen mainScreen].maximumFramesPerSecond > 60;
#endif // !TARGET_OS_OSX
```